### PR TITLE
[FIX] Share Extension issues

### PR DESCRIPTION
**Pull request by djorkaeffalexandre**
Share Extension iOS crashes with approximately 500 chats, we will go to slice chats with limit 50 and show the fifty most recents only.
This pr also adds header search and fixes some bugs on android because on android the store is shared between App and Share Extension when both are open.

### CODE REVIEW

File: app/views/ShareListView/index.js [hunk 1]
```diff
@@ -308,7 +269,7 @@ export default class ShareListView extends React.Component {
 	search = (text) => {
 		const result = database.objects('subscriptions').filtered('name CONTAINS[c] $0', text);
 		const subscriptions = database.objects('subscriptions');
-		const data = result.length !== subscriptions.length ? result : [];
+		const data = result.length !== subscriptions.length ? result.slice(0, 50) : [];
```
Comment (diegolmello) [inline app/views/ShareListView/index.js]: If search length === 0, show a label.

Comment (djorkaeffalexandre) [reply app/views/ShareListView/index.js]: fixed.

File: app/lib/realm.js [hunk 2]
```diff
@@ -424,6 +427,16 @@ class DB {
 						newServers[i].roomsUpdatedAt = null;
 					}
 				}
+				if (oldRealm.schemaVersion === 9 && newRealm.schemaVersion === 10) {
```
Comment (diegolmello) [inline app/lib/realm.js]: Not needed

File: app/views/ShareListView/SearchBar.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+	const onCancelPress = () => {
+		Keyboard.dismiss();
+		onChangeText('');
+		setHasCancel(false);
+	};
+
+	const onFocus = () => {
+		setHasCancel(true);
+		LayoutAnimation.easeInEaseOut();
```
Comment (diegolmello) [inline app/views/ShareListView/SearchBar.js]: Wrap with `isIOS` condition

File: app/views/ShareView/index.js [hunk 4]
```diff
@@ -112,8 +114,10 @@ export default class ShareView extends React.Component {
 
 	sendTextMessage = async() => {
 		const { value, rid } = this.state;
+		const { token } = this.props;
 		if (value !== '' && rid !== '') {
 			try {
+				await RocketChat.login({ resume: token });
```
Comment (diegolmello) [inline app/views/ShareView/index.js]: Shouldn't you do the same for `sendMediaMessage`?
Why not login in `shareExtensionInit`?

File: app/views/ShareListView/index.js [hunk 5]
```diff
@@ -22,31 +21,27 @@ import {
...
-import ServerItem, { ROW_HEIGHT as ROW_HEIGHT_SERVER } from '../../presentation/ServerItem';
+import ServerItem from '../../presentation/ServerItem';
 import { CloseShareExtensionButton, CustomHeaderButtons, Item } from '../../containers/HeaderButton';
-import SearchBar from '../RoomsListView/ListHeader/SearchBar';
+import SearchBar from './SearchBar';
 import ShareListHeader from './Header';
 
 import styles from './styles';
 
 const SCROLL_OFFSET = 56;
```
Comment (diegolmello) [inline app/views/ShareListView/index.js]: It never hides.
Can you place the searchbox at the header?

Comment (djorkaeffalexandre) [reply app/views/ShareListView/index.js]: done.

File: app/views/ShareListView/Header.js [hunk 6]
```diff
@@ -84,6 +85,12 @@ class ShareListHeader extends PureComponent {
 					/>
 				</View>
 			);
+		} else if (isIOS) {
```
Comment (diegolmello) [inline app/views/ShareListView/Header.js]: Why do we have a `SearchHeader` component for iOS and a TextInput on Android?
It should be the component, if necessary.

File: app/views/ShareListView/SearchHeader.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+import { HEADER_BACKGROUND } from '../../constants/colors';
+
+import sharedStyles from '../Styles';
+
+const styles = StyleSheet.create({
+	container: {
+		backgroundColor: HEADER_BACKGROUND,
+		width: '100%',
+		flexDirection: 'row',
+		paddingTop: getStatusBarHeight() + (isNotch ? 20 : 0),
```
Comment (diegolmello) [inline app/views/ShareListView/SearchHeader.js]: I didn't like this lib.
I think we probably don't need it.
I'll take a look.

File: app/views/ShareView/index.js [hunk 8]
```diff
@@ -81,6 +85,11 @@ export default class ShareView extends React.Component {
 		navigation.setParams({ sendMessage: this._sendMessage, canSend: !(isReadOnly(room, { username }) || isBlocked(room)) });
 	}
 
+	componentDidUpdate() {
+		const { setSearch } = this.props;
+		setSearch('');
```
Comment (diegolmello) [inline app/views/ShareView/index.js]: Bad practice.
You executing `setSearch('')` for every render.
Is this what you want to do here?

**Comment by djorkaeffalexandre** [comment]
@diegolmello Apple News was fixed on `rn-extensions-share v2.3.8`.

File: app/lib/methods/sendMessage.js [hunk 9]
```diff
@@ -5,8 +5,9 @@ import reduxStore from '../createStore';
 import log from '../../utils/log';
 import random from '../../utils/random';
 
-export const getMessage = (rid, msg = '', tmid) => {
+export const getMessage = (rid, msg = '', tmid, shareExtension) => {
```
Comment (diegolmello) [inline app/lib/methods/sendMessage.js]: This is not scalable.
You can send `userId` and `username` as params instead.

Comment (djorkaeffalexandre) [reply app/lib/methods/sendMessage.js]: Fixed.

**Pull request merged**